Clean and refactor sesameService test

s-petit opened the pull request:
We had formatting issues and used deprecated code (TestBed.get)

This commit cleans the test.

s-petit committed:
Clean and refactor sesameService test

We had formatting issues and used deprecated code (TestBed.get)

This commit cleans the test.

AmauryBchp commented on src/lib/sesame.service.spec.ts:
```diff
@@ -34,58 +30,54 @@ describe('SesameService', () => {
-      expect(req.request.headers.get('Authorization')).toBe('Basic ' + btoa('toto:password'));
-    })
-  );
+  it('should retrieve UserInfo', () => {
+    sesame.userInfo().subscribe((userInfo) => {
+        expect(userInfo.username).toEqual('toto');
+      }
+    );
+    flushTestData();
+  });
+
+  it('should has role ROLE1', () => {
+    sesame.hasRole('ROLE1').subscribe((hasRole) =>
+      expect(hasRole).toBeTruthy()
+    );
+    flushTestData();
+  });
+
+  it('should has one role NOT_AVAILABLE,ROLE1', () => {
+    sesame.hasAnyRoles(['NOT_AVAILABLE', 'ROLE1']).subscribe((hasRole) =>
+      expect(hasRole).toBeTruthy()
+    );
+    flushTestData();
+  });
```
Je chipote mais vu que c'est des tests qui se ressemblent, peut être mettre dans un describe et un before each pour éviter de dupliquer flushTestData mais bon c'est pas très important

s-petit replied:
Bonne idée, je fais ca :+1:

AmauryBchp reviewed (approved):
go :rocket:

s-petit committed:
Improve sesame service test

We splitted the test in two parts :

- When the user is already logged via the cookie
- When the user is not logged

Also, we cleaned the Http verify assertions in a dedicated test.

Moreover, we added a new rule: We do not want to call sesame server when user is not logged and not have any cookie. It will prevent errors and unwanted http calls.

AmauryBchp commented on src/lib/sesame.service.spec.ts:
```diff
@@ -12,6 +12,7 @@ const userInfoStub = {
 } as UserInfo;
 
 const jwtToken = 'dummy.' + btoa(JSON.stringify(userInfoStub)) + '.dummy';
+const cookie = 'authentication-jwt=dummy.eyJ1c2VybmFtZSI6InRvdG8iLCJtYWlsIjoidG90b0B2aWRhbC5mciIsImxhbmd1YWdlIjp7ImNvZGUiOiJmcmEiLCJsYWJlbCI6IkZyYW5jYWlzIn0sInJvbGVzIjpbIlJPTEUxIiwiUk9MRTIiXX0=.dummy'
```
Petit ; à la fin de la ligne :innocent:

AmauryBchp reviewed (approved):
Niquel ces tests :+1:
